src/Http/Http/src/DefaultHttpContext.cs

```diff
@@ -213,6 +214,34 @@ public override ISession Session
         }
     }
 
+    /// <summary>
+    /// Gets or sets the <see cref="Endpoint"/> for this request.
+    /// </summary>
+    public override Endpoint? Endpoint
+    {
+        get
+        {
+            var endpointFeature = Features.Get<IEndpointFeature>();
+            if (endpointFeature == null)
```

Comment:
I think this is _technically_ a breaking behavioral change since the feature is set on the context even if there is no endpoint. This could pose an issue for codepaths that assumed that no `IEndpointFeature` == no endpoint. 

From looking at usages of this API ([ref](https://grep.app/search?current=2&q=.Get%3CIEndpointFeature%3E&filter[lang][0]=C%23)), this might not functionally be a problem but good to note.